server/routes/users.js

```diff
@@ -2,67 +2,101 @@ const { v4: uuidv4 } = require('uuid');
 var express = require("express");
 var router = express.Router();
 
-var {mongoose, db, User} = require('../database');
+// var {mongoose, db, User: UserModel} = require('../database/database');
+var db = require('../DatabaseAccesser')
 
 router.get('/', function(req, res, next) {
-    User.find(function(err, users){
-        if(err) {
-            console.log(err);
-        } else {
-            res.json(users);
-        }
-    })
+
+    db.getAllUsers()
+        .then((users)=>{
+            if (users) {
+                res.status(200).json(users)
+            }
+            else {
+                res.status(500)
+            }
+        })
 })
 
 router.post('/teacher/', function(req, res, next) {
-    const emailQuery = User.where({email: req.body.email});
-    var user = new User({
-        name: req.body.displayName,
-        email: req.body.email,
-        password: req.body.password,
-        displayName: req.body.displayName,
+    var user = {
```

Comment:
A request's json is stored at req.body, not req. i.e. (req.email => req.body.email)